### PR TITLE
ICDS Dashboard: Additional filters for mobile dashboard

### PR DESCRIPTION
**Pull request by saiprakash-v**
https://trello.com/c/1iJz4TF2
##### SUMMARY

Added additional filter template and tests for mobile dashboard

### CODE REVIEW

File: custom/icds_reports/static/icds_reports/js/spec/additional-filter.directive.spec.js
Comment (czue) [inline custom/icds_reports/static/icds_reports/js/spec/additional-filter.directive.spec.js]: I think this can use `provideDefaultConstants` and pass in the appropriate options to include these

File: custom/icds_reports/static/js/angular-services/baseControllers.service.js
Comment (czue) [inline custom/icds_reports/static/js/angular-services/baseControllers.service.js]: nit: I think it'd be slightly nicer if it only added these if necessary. either based on whether values exist or the report is setup to include them. also if you did that it might be good to rename the method to `addAdditionalFiltersIfNecessary`

File: custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js
Comment (czue) [reply custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js]: can you fix this one?

File: custom/icds_reports/static/js/directives/mobile-filters/mobile-filters.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/mobile-filters/mobile-filters.directive.js]: I find parsing this logic quite difficult, possibly because I don't quite understand the variable name and also the conditional is a little odd. I think it's: `!(genderFilterExists || ageFilterExists)`, yeah? So if either of them exist then `receivedAdditionalFilterData` `false`? Can you explain why that would be the right logic (possibly with a comment?).

File: custom/icds_reports/static/js/directives/mobile-filters/mobile-filters.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/mobile-filters/mobile-filters.directive.js]: `receivedAdditionalFilterData`  will always be `true` even if the report doesn't have additional filters? can you explain how that works? it's a bit counter-intuitive to me (again, maybe just a non-intuitive variable name causing the issue)

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/additional-filter.html
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/additional-filter.html]: having now seen this more than once maybe make it a variable or function on the controller so it's more readable?

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/additional-filter.html
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/additional-filter.html]: same

File: custom/icds_reports/templates/icds_reports/partials/shared_component_directives.html
Comment (czue) [inline custom/icds_reports/templates/icds_reports/partials/shared_component_directives.html]: surprised to see this here without it also being removed from somewhere. did you forget to delete it from `dashboard.html` and `mocha.html`?

**Review by czue — CHANGES_REQUESTED** [review]
Looking good! Mostly small-ish stuff and questions. Also, can you please post a screenshot and add a bit more detail on how this works to the description?

File: custom/icds_reports/static/js/directives/mobile-filters/mobile-filters.directive.js
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/mobile-filters/mobile-filters.directive.js]: Sure I will add code comment

File: custom/icds_reports/templates/icds_reports/partials/shared_component_directives.html
Comment (saiprakash-v) [reply custom/icds_reports/templates/icds_reports/partials/shared_component_directives.html]: Should I add this to shared components? (as in which components are to be included here?). Just because other filter directives were added, I added this too. (Waiting for review)

Comment (czue) [reply custom/icds_reports/templates/icds_reports/partials/shared_component_directives.html]: being here means it's used in both mobile and web dashboard (which is correct), but once that's done we should remove it explicitly from web and tests since those already import this whole file. does that make sense?

Comment (saiprakash-v) [reply custom/icds_reports/templates/icds_reports/partials/shared_component_directives.html]: Got it.

File: custom/icds_reports/static/js/angular-services/baseControllers.service.js
Comment (saiprakash-v) [reply custom/icds_reports/static/js/angular-services/baseControllers.service.js]: I'm not sure of the reason. But this code already does the same job. For reports where no filters should be shown, gender and age is removed from url.

Comment (czue) [reply custom/icds_reports/static/js/angular-services/baseControllers.service.js]: Great! :man_shrugging:

**Comment by saiprakash-v** [comment]
> Looking good! Mostly small-ish stuff and questions. Also, can you please post a screenshot and add a bit more detail on how this works to the description?

Screenshots:
Screenshot from 2020-02-13 18-41-04
Screenshot from 2020-02-13 18-41-32
Screenshot from 2020-02-13 18-42-10
Screenshot from 2020-02-13 18-42-37

File: custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js]: should these be initialized to `false`? (I realize `undefined` also works, but think it's clearer to be explicit)

Comment (czue) [reply custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js]: the other option is to have
```javascript
vm.showGenderFilter = function() {
  return vm.filters && vm.filters.indexOf('gender') === -1;
}
```
which was what I was thinking of originally

Comment (czue) [reply custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js]: (both of these are optional IMO but will let you decide if you want to do either one)

**Review by czue — APPROVED** [review]
Looks good. Just waiting to confirm with Saket whether we want to put this behind a flag

File: custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js]: I'm going to initialize variables to false. But will let other things stay as is.

**Review by czue — APPROVED** [review]

**Comment by czue** [comment]
we're going to merge this after tonights deploy

**Comment by czue** [comment]
thanks @saiprakash-v

**Pull request merged**
